web/app/scripts/utils/date-utils.js

```diff
@@ -0,0 +1,152 @@
+(function() {
+    'use strict';
+
+    /**
+     * Date.parse with progressive enhancement for ISO 8601 <https://github.com/csnover/js-iso8601>
+     * NON-CONFORMANT EDITION.
+     * © 2011 Colin Snover <http://zetafleet.com>
+     * Released under MIT license.
+     * Using this for IE8 compatability
+     */
+    (function(Date, undefined) {
+        var origParse = Date.parse,
+            numericKeys = [1, 4, 5, 6, 10, 11];
+        Date.parse = function(date) {
+            var timestamp, struct, minutesOffset = 0;
+
+            //              1 YYYY                 2 MM        3 DD              4 HH     5 mm        6 ss            7 msec         8 Z 9 ±    10 tzHH    11 tzmm
+            if ((struct = /^(\d{4}|[+\-]\d{6})(?:-?(\d{2})(?:-?(\d{2}))?)?(?:[ T]?(\d{2}):?(\d{2})(?::?(\d{2})(?:[,\.](\d{1,}))?)?(?:(Z)|([+\-])(\d{2})(?::?(\d{2}))?)?)?$/.exec(date))) {
+                // avoid NaN timestamps caused by “undefined” values being passed to Date.UTC
+                for (var i = 0, k;
+                    (k = numericKeys[i]); ++i) {
+                    struct[k] = +struct[k] || 0;
+                }
+
+                // allow undefined days and months
+                struct[2] = (+struct[2] || 1) - 1;
+                struct[3] = +struct[3] || 1;
+
+                // allow arbitrary sub-second precision beyond milliseconds
+                struct[7] = struct[7] ? +(struct[7] + "00").substr(0, 3) : 0;
+
+                // timestamps without timezone identifiers should be considered local time
+                if ((struct[8] === undefined || struct[8] === '') && (struct[9] === undefined || struct[9] === '')) {
+                    timestamp = +new Date(struct[1], struct[2], struct[3], struct[4], struct[5], struct[6], struct[7]);
+                } else {
+                    if (struct[8] !== 'Z' && struct[9] !== undefined) {
+                        minutesOffset = struct[10] * 60 + struct[11];
+
+                        if (struct[9] === '+') {
+                            minutesOffset = 0 - minutesOffset;
+                        }
+                    }
+
+                    timestamp = Date.UTC(struct[1], struct[2], struct[3], struct[4], struct[5] + minutesOffset, struct[6], struct[7]);
+                }
+            } else {
+                timestamp = origParse ? origParse(date) : NaN;
+            }
+
+            return timestamp;
+        };
+    }(Date));
+
+    /* ngInject */
+    function DateUtils(LanguageState) {
+        var languageMap = {
```

Comment:
As a thought, would it make sense to move this into `config.js` inside the objects that hold the rest of the language data? That way all the language-specific data would be packaged up together and it would be easy to add new date formats by just editing that config.